### PR TITLE
feat: add dynamic secret masking doc

### PR DESCRIPTION
**Pull request by eliecharra**
# Description of the change

## Checklist

Please make sure that the proposed change checks all the boxes below before requesting a review:

- [x] I have reviewed the [guidelines for contributing](https://github.com/spacelift-io/user-documentation/blob/main/CONTRIBUTING.md) to this repository.
- [x] The preview looks fine.
- [x] The tests pass.
- [x] The commit history is clean and meaningful.
- [x] The pull request is opened against the `main` branch.
- [x] The pull request is no longer marked as a draft.
- [ ] You agree to license your contribution under the [MIT license](https://github.com/spacelift-io/user-documentation/blob/main/LICENSE) to Spacelift (not required for Spacelift employees).
- You have updated the navigation files correctly:
    - [ ] No new pages have been added, or;
    - [ ] Only _nav.yaml_ has been updated because the changes only apply to SaaS, or;
    - [ ] Only _nav.self-hosted.yaml_ has been updated because the changes only apply to Self-Hosted, or;
    - [ ] Both _nav.yaml_ and _nav.self-hosted.yaml_ have been updated.

If the proposed change is ready to be merged, please request a review from `@spacelift-io/solutions-engineering`. Someone will review and merge the pull request.

_Spacelift employees should request reviews from the relevant engineers and are allowed to merge pull requests after they got at least one approval._

Thank you for your contribution! 🙇

### CODE REVIEW

File: docs/concepts/stack/stack-settings.md [hunk 1]
```diff
@@ -66,6 +66,40 @@ Environment variables are preserved from one phase to the next.
...
+
+You could use those commands in any lifecycle step of the workflow.
+
+![stack_runtime_command](../../assets/screenshots/stack_runtime_command.png)
+
+```bash
+echo "::command arg1 arg2"
+```
+
+Below is a list of supported commands. See more detailed doc after the table.
```
Comment (adamconnelly) [inline docs/concepts/stack/stack-settings.md:82]:
```suggestion
Below is a list of supported commands. See the more detailed doc after this table.
```

File: docs/concepts/stack/stack-settings.md [hunk 2]
```diff
@@ -66,6 +66,40 @@ Environment variables are preserved from one phase to the next.
...
+
+| Command                   | Description                                    |
+|---------------------------|------------------------------------------------|
+| [`::add-mask`](#add-mask) | Allow to set values to be masked in the output |
```
Comment (adamconnelly) [inline docs/concepts/stack/stack-settings.md:86]:
```suggestion
| [`::add-mask`](#add-mask) | Adds a set of values that should be masked in log output |
```

File: docs/concepts/stack/stack-settings.md [hunk 3]
```diff
@@ -66,6 +66,40 @@ Environment variables are preserved from one phase to the next.
...
+
+#### ::add-mask
+
+When you mask a value, it is treated as a secret and will be redacted in the logs output.
+Each masked word separated by whitespace is replaced with five * character.
```
Comment (adamconnelly) [inline docs/concepts/stack/stack-settings.md:91]:
```suggestion
Each masked word separated by whitespace is replaced with five `*` characters.
```

File: docs/concepts/stack/stack-settings.md [hunk 4]
```diff
@@ -66,6 +66,40 @@ Environment variables are preserved from one phase to the next.
...
+
+##### Example
+
+```bash
+# Multiples masks can be set with a single command
```
Comment (adamconnelly) [inline docs/concepts/stack/stack-settings.md:96]:
```suggestion
# Multiple masks can be set with a single command
```

**Review by adamconnelly — COMMENTED** [review]
Just some minor tweaks.

**Comment by eliecharra** [comment]
Ha cool thanks @adamconnelly 🙏🏻 
I was planning to do a second pass after putting it out of draft but you dit it already 😅 🚀

**Comment by adamconnelly** [comment]
> Ha cool thanks @adamconnelly 🙏🏻 I was planning to do a second pass after putting it out of draft but you dit it already 😅 🚀

Ah right, sorry. It was just because you'd linked it from the backend PR so I figured I'd take a look.

**Comment by eliecharra** [comment]
Ah yeah fair enough 🙏🏻

**Review by peterdemespacelift — APPROVED** [review]

**Review by peterdeme — APPROVED** [review]

**Pull request merged**